tests/push/test_push_rules_http.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2018 New Vector
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from unittest.mock import Mock
+
+from twisted.internet.defer import Deferred
+
+import synapse.rest.admin
+from synapse.logging.context import make_deferred_yieldable
+from synapse.rest.client import login, push_rule, receipts, room
+
+from tests.unittest import HomeserverTestCase
+
+
+class HTTPPusherTests(HomeserverTestCase):
```

Comment:
Not a requirement, but if this and `HTTPPusherTests` share a lot of code (or could) it can be nice to have an abstract base class they both inherit from.

It really isn't 100% clear to me why this is a different class anyway? Should this be renamed or maybe have comments added about who they differ?

Reply:
Totally agree - was originally separate for ease of writing / testing it, I've merged it back into the existing module in https://github.com/matrix-org/synapse/pull/12188/commits/26121c719c873ff74e912c69ac8edf59380a1e5b.